packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -145,12 +162,67 @@ export class Bootstrapper {
         CloudFormationExecutionPolicies: cloudFormationExecutionPolicies.join(','),
         Qualifier: params.qualifier,
         PublicAccessBlockConfiguration: params.publicAccessBlockConfiguration || params.publicAccessBlockConfiguration === undefined ? 'true' : 'false',
+        PermissionsBoundary: policyName,
       }, {
         ...options,
         terminationProtection: options.terminationProtection ?? current.terminationProtection,
       });
   }
 
+  private async getPolicyName(
+    environment: cxapi.Environment,
+    sdk: ISDK,
+    permissionsBoundary: string,
+    template: string,
+    partition: string,
+    params: BootstrappingParameters): Promise<string> {
+
+    if (permissionsBoundary !== CDK_BOOTSTRAP_PERMISSIONS_BOUNDARY) {
+      this.validatePolicyName(permissionsBoundary);
+      return Promise.resolve(permissionsBoundary);
+    }
+    // if no Qualifier is supplied, resort to the default one
+    const arn = await this.getExamplePermissionsBoundary(template, params.qualifier ?? 'hnb659fds', partition, environment.account, sdk);
+    const policyName = arn.split('/').pop();
+    if (!policyName) {
+      throw new Error('Could not retrieve the example permission boundary!');
+    }
+    return policyName;
+  }
+
+  private async getExamplePermissionsBoundary(template: string, qualifier: string, partition: string, account: string, sdk: ISDK): Promise<string> {
+    const iam = sdk.iam();
+
+    let policyName = `cdk-${qualifier}-permissions-boundary`;
+    const arn = `arn:${partition}:iam::${account}:policy/${policyName}`;
+
+    let getPolicyResp = await iam.getPolicy({ PolicyArn: arn }).promise();
+    if (getPolicyResp.Policy) {
+      return arn;
+    }
+
+    const policyDoc = JSON.parse(serializeStructure(template, true)).CdkBoostrapPermissionsBoundaryPolicy;
```

Comment:
```suggestion
    const policyDoc = JSON.parse(serializeStructure(template, true)).Resources.CdkBoostrapPermissionsBoundaryPolicy.Properties.PolicyDocument;
```

Also, looks like the policy is valid from CloudFormations perspective because it
contains CloudFormation intrinsics which doesn't work with the IAM API. Not sure
if it would be easier to do a Regex replacement or just manage the policy in two
	places? I would be fine managing it in two places if that is easier.

Reply:
>do a Regex replacement

sounds more heavy-duty -- can I open a follow-up for that POC and switch back to the policy doc as string for now?

Reply:
> sounds more heavy-duty -- can I open a follow-up for that POC and switch back to the policy doc as string for now?

Yeah sounds good to me